near-sdk/src/store/free_list/mod.rs

```diff
@@ -223,6 +223,92 @@ where
     pub fn drain(&mut self) -> Drain<T> {
         Drain::new(self)
     }
+
+    /// Empty slots in the front of the list is swapped with occupied slots in back of the list.
+    /// Defrag helps reduce gas cost in certain scenarios where lot of elements in front of the list are
+    /// removed without getting replaced. Please see https://github.com/near/near-sdk-rs/issues/990
+    pub fn defrag<F>(&mut self, callback: F)
```

Comment:
Might make sense to have the callback be optional? In most cases I would assume devs won't care about how the indices change. That or make this method `pub(crate)` to ensure that if this type (`FreeList`) becomes public, this API doesn't get exposed as such.